### ISSUE TITLE
Camelot Key on S4mk3

### ISSUE THREAD

**Issue by yuletide** [kokernutz/traktor-kontrol-screens#29]
Hey folks,
Just got an s4mk3 and was sent here by another fan. I'm very new to QT but managed to get Camelot keys to show, but not while respecting the preference setting -- I'm guessing because of the nested if's. Is there a better way to do this?

` text: deckInfo.hasKey ? (prefs.camelotKey ? utils.convertToCamelot(deckInfo.keyString) : deckInfo.keyString) : "No key"`

When I take out the preference check it works fine:

https://github.com/yuletide/traktor-kontrol-screens/commit/482a88f75a8e83b51862df1500e353e56f0e11aa

**Comment by kokernutz** [comment]
Heya! 

Nice work!

I ordered an S4 MK3 and it's getting delivered on Wednesday. I was going to update my mods so it displays Camelot keys on the new controller. Can you wait until then?

**Comment by yuletide** [comment]
@kokernutz Of course! I also added in setting key text color based on your utility functions which looks great. Excited to see what else we can do with these little screens.

Unrelated, but have you any experience creating mappings? My next attempt was to somehow change master tempo (there's no knob), but I think we can do that without messing about in qml

**Comment by kokernutz** [comment]
Yes, I made a keyboard mapping so I could prep tracks quickly.

Doesn't SHIFT-TEMPO FADER do that? I had an S4 MK2 years ago and vaguely remember that.

**Comment by yuletide** [comment]
So it seems shift-fader does `literally nothing` which is actually useful since you can reset the fader position to neutral without sending any signal. I just discovered it this last second which solved my problem of running out of fader room when set to `relative` and loading a new track with the fader already down

**Comment by kokernutz** [comment]
Ah yes, that's it. Still need a mapping to fix master tempo?
